### PR TITLE
fix: don't create new instances of builders classes

### PR DESCRIPTION
**Pull request by ImRodry**
**Please describe the changes this PR makes and why it should be merged:**

This PR makes it so that if a builders class is passed onto createComponent it is returned as-is, without creating a new instance of it so we can keep referencing it. Also changed types of ActionRow slightly to make it easier to type it

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes

### CODE REVIEW

File: packages/builders/src/components/ActionRow.ts [hunk 1]
```diff
@@ -3,14 +3,16 @@ import type { ButtonComponent, SelectMenuComponent } from '..';
...
+
 export type ActionRowComponent = ButtonComponent | SelectMenuComponent;
 
 // TODO: Add valid form component types
 
 /**
  * Represents an action row component
  */
-export class ActionRow<T extends ActionRowComponent> implements Component {
+export class ActionRow<T extends ActionRowComponent = ActionRowComponent> implements Component {
```
Comment (suneettipirneni) [inline packages/builders/src/components/ActionRow.ts:15]: Out of scope?

Comment (ImRodry) [reply packages/builders/src/components/ActionRow.ts:15]: Eh maybe? I explained in the description it was just a small thing to make it easier to type but I can revert if you want

File: packages/builders/src/components/Components.ts [hunk 2]
```diff
@@ -15,14 +15,15 @@ export interface MappedComponentTypes {
 export function createComponent<T extends keyof MappedComponentTypes>(
 	data: APIMessageComponent & { type: T },
 ): MappedComponentTypes[T];
-export function createComponent(data: APIMessageComponent): Component {
+export function createComponent<C extends MessageComponent>(data: C): C;
+export function createComponent(data: APIMessageComponent | MessageComponent): Component {
```
Comment (vladfrangu) [inline packages/builders/src/components/Components.ts:19]: Shouldn't you just.. type it as `data: APIMessageComponent | Component`?

**Review by vladfrangu — DISMISSED** [review]

File: packages/builders/src/components/Components.ts (hunk 2 above)
Comment (ImRodry) [reply packages/builders/src/components/Components.ts:19]: can't
image

**Review by iCrawl — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**